From 97d8ba9e04596c64fbf2e86740926f6e1a413ebf Mon Sep 17 00:00:00 2001
From: tgeisler <tgeisler@90b65887-3827-0410-9a23-83215b262276>
Date: Tue, 24 Oct 2017 14:06:01 +0000
Subject: [PATCH 0812/1000] MANTIS 0048136 0048136: CfgValues: command
 REMOVEFROMLIST does not remove values if escaped chars are included

git-svn-id: svn://localhost/SelfServiceCommon/trunk@1281 90b65887-3827-0410-9a23-83215b262276
---
 .../Massai/cpp/CfgTools/inc/CfgValues_def.h   |  9 ++-
 .../Massai/cpp/CfgTools/src/CfgValues.cpp     | 72 +++----------------
 2 files changed, 17 insertions(+), 64 deletions(-)

diff --git a/SelfServiceCommon/Massai/cpp/CfgTools/inc/CfgValues_def.h b/SelfServiceCommon/Massai/cpp/CfgTools/inc/CfgValues_def.h
index 15eb4575..a7fd9f43 100644
--- a/SelfServiceCommon/Massai/cpp/CfgTools/inc/CfgValues_def.h
+++ b/SelfServiceCommon/Massai/cpp/CfgTools/inc/CfgValues_def.h
@@ -79,8 +79,13 @@
 //                        if syntax of single item config is wrong
 //                      - Generate a line-break warning, if one is detected
 //                        within an input record.
-#define FILE_VER_BIN 1,1,5,0
-#define FILE_VER_DTS "1.1.5.0\0"
+//#define FILE_VER_BIN 1,1,6,0
+//#define FILE_VER_DTS "1.1.6.0\0"
+//
+// 24.10.2017  GiT      - Mantis 0048136: CfgValues: command REMOVEFROMLIST does not remove values if escaped chars are included
+//
+#define FILE_VER_BIN 1,1,6,0
+#define FILE_VER_DTS "1.1.6.0\0"
 
 //-------------------------------------------------------------------------
 
diff --git a/SelfServiceCommon/Massai/cpp/CfgTools/src/CfgValues.cpp b/SelfServiceCommon/Massai/cpp/CfgTools/src/CfgValues.cpp
index a2da2f2f..7e188f61 100644
--- a/SelfServiceCommon/Massai/cpp/CfgTools/src/CfgValues.cpp
+++ b/SelfServiceCommon/Massai/cpp/CfgTools/src/CfgValues.cpp
@@ -41,34 +41,16 @@
 #define HINDENT1        "          "
 #define HINDENT2        "                           "
 
-
-
 namespace
 {
-   
-   
    int const c_exit_good = 0;
-   
-   
    int const c_exit_some_items_failed = 1;
-   
-   
    int const c_exit_rejected = 2;
-
-   
    int const c_exit_connection_failed = 3;
-   
-   
    int const c_exit_wrong_parameter = 4;
-   
-   
    int const c_exit_file_not_read = 5;
-   
-   
 } // namespace
 
-
-
 /**
  *
  * Generate ATSAVER string.
@@ -80,8 +62,6 @@ namespace
 
 char _atsaver[] = MAT_VERS_ID FILE_VER_DTS;
 
-
-
 /**
  *
  * Write the item prolog.
@@ -99,8 +79,6 @@ void anyItemFirst
          " val=" CSTR_VAL_BEG;
 }
 
-
-
 /**
  *
  * Write the item epilog.
@@ -114,8 +92,6 @@ void anyItemLast
    OS << CSTR_VAL_END << std::endl;
 }
 
-
-
 /**
  *
  * Copy one cfg-format text-element to an ostream.
@@ -142,8 +118,6 @@ void readTextCopy( std::ostream & OS, char const * Src )
    }
 }
 
-
-
 /**
  *
  * Copy one cfg-format list-element to an ostream.
@@ -171,8 +145,6 @@ void readTextListCopy( std::ostream & OS, char const * Src )
    }
 }
 
-
-
 /**
  *
  * Copy from flat to string.
@@ -211,8 +183,6 @@ void writeTextCopy( std::string & DstVal, char const * Src )
    }
 }
 
-
-
 /**
  *
  * Copy one list-element from flat to a cfg-format destination buffer.
@@ -265,8 +235,6 @@ char const * writeTextListCopyN
    }
 }
 
-
-
 /**
  *
  * Copy from flat to string list.
@@ -310,8 +278,6 @@ bool writeTextListCopy
    return FormatError;
 }
 
-
-
 /**
  *
  * Read values from the configuration base to the ostream specified.
@@ -330,7 +296,6 @@ void cfgReadDirectory
    Acc.Item = CORBA::string_dup( ParaPath.c_str() );
    Acc.Authorization = CORBA::string_dup( Auth );
 
-
    try
    {
       Config::Directory::dir * const Dir = DirProxy->get( Acc );
@@ -464,8 +429,6 @@ void cfgReadDirectory
 
 }
 
-
-
 /**
  *
  * cfgValuesFRead() execution status.
@@ -480,10 +443,8 @@ struct cfgValuesFStatus
       m_newlines( 0 )
    {
    }
-   
-};
-
 
+};
 
 /**
  *
@@ -585,8 +546,6 @@ void cfgValuesFRead
    return;
 }
 
-
-
 /**
  *
  * Read the next token value if token is found, do nothing otherwise
@@ -686,8 +645,6 @@ void cfgValuesFReadE
    return;
 }
 
-
-
 /**
  *
  * Read the next token value if token is found, do nothing otherwise
@@ -698,8 +655,6 @@ void ignoredArg( char const * Param )
    std::cout << "Parameter [" << Param << "] ignored!" << std::endl;
 }
 
-
-
 /**
  *
  * Stream out the number of items.
@@ -716,8 +671,6 @@ void soutItems( unsigned N )
       std::cout << "s";
 }
 
-
-
 /**
  *
  * Retrieve the value
@@ -1497,7 +1450,7 @@ int main( int argc, char ** argv )
         Config::description CDescr;
         SAcc.Authorization = CORBA::string_dup( ACC_NAME );
         SAcc.Item = CORBA::string_dup( RemItem );
-            
+
         corbaClient<Config::Textlist>
           TextlistClnt( Host, CFGMGR_PORT, CFGMGR_SRVNAME_TEXTLIST );
         Config::Textlist * const TextlistProxy = TextlistClnt.proxy();
@@ -1516,34 +1469,31 @@ int main( int argc, char ** argv )
             Config::Textlist::strings const & Val =
                *TextlistProxy->get( SAcc );
             unsigned int Length = Val.length();
-            std::stringstream strbuf;
+            std::string strbuf;
             Config::Textlist::strings cstrbuflist;
             cstrbuflist.length(Length);
             std::cout
-               << "Found " << Length << " Textlistitems"
+               << "Found " << Length << " Textlistitems" << " compare with [" << RemValue << "]"
                << std::endl;
-            
+
             int IdxNew = 0;
             for( unsigned Idx = 0; Idx < Length; ++Idx )
             {
-               readTextListCopy( strbuf, Val[Idx].in() );
+               strbuf.assign(Val[Idx].in());
 
-               if( strbuf.str().compare(RemValue) != 0 )
+               if( strbuf.compare(RemValue) != 0 )
                {
                  cstrbuflist[IdxNew++]=(CORBA::string_dup(Val[Idx].in()));
                }
-               strbuf.str(""); strbuf.clear();
+               strbuf.clear();
             }
-            if( IdxNew != Length && ( IdxNew > 0 ) )
+            if( IdxNew != Length )
             {
                cstrbuflist.length(IdxNew);
                try
                {
                   TextlistProxy->set( SAcc, cstrbuflist );
-                  std::cout
-                     << "Removed [" << RemValue << "]"
-                        " from Listitem [" << RemItem << "] "
-                     << std::endl;
+                  std::cout << "Removed [" << RemValue << "] from Listitem [" << RemItem << "] " << std::endl;
                }
                catch( ... )
                {
@@ -1661,8 +1611,6 @@ int main( int argc, char ** argv )
    return c_exit_rejected;
 }
 
-
-
 /**
  **************************************************************************
  *
-- 
2.41.0.windows.1

